### Import Library

In [1]:
import pandas as pd
import numpy as np
import matplotlib as mat
from scipy import stats

### Read data csv

In [2]:
df = pd.read_csv("data/covid_19_indonesia_time_series_all.csv")
df

Date Location ISO Code          Location  New Cases  New Deaths  \
0       3/1/2020             ID-JK       DKI Jakarta          2           0   
1       3/2/2020             ID-JK       DKI Jakarta          2           0   
2       3/2/2020               IDN         Indonesia          2           0   
3       3/2/2020             ID-RI              Riau          1           0   
4       3/3/2020             ID-JK       DKI Jakarta          2           0   
...          ...               ...               ...        ...         ...   
31817  9/15/2022             ID-SA    Sulawesi Utara         37           0   
31818  9/15/2022             ID-SB    Sumatera Barat         13           0   
31819  9/15/2022             ID-SS  Sumatera Selatan         16           0   
31820  9/15/2022             ID-SU    Sumatera Utara         50           0   
31821  9/16/2022               IDN         Indonesia       2358          27   

       New Recovered  New Active Cases  Total Cases  Total Deaths  \
0                  0                 2           39            20   
1                  0                 2           41            20   
2                  0                 2            2             0   
3                  0                 1            1             0   
4                  0                 2           43            20   
...              ...               ...          ...           ...   
31817              0                37        52770          1213   
31818              3                10       104640          2371   
31819              1                15        82198          3376   
31820              5                45       158866          3288   
31821           2997              -666      6405044        157876   

       Total Recovered  ...  Latitude New Cases per Million  \
0                   75  ... -6.204699                  0.18   
1                   75  ... -6.204699                  0.18   
2                    0  ... -0.789275                  0.01   
3                    1  ...  0.511648                  0.16   
4                   75  ... -6.204699                  0.18   
...                ...  ...       ...                   ...   
31817            50997  ...  1.259638                 14.01   
31818           102066  ... -0.850253                  2.36   
31819            78510  ... -3.216212                  1.95   
31820           154924  ...  2.191894                  3.36   
31821          6218708  ... -0.789275                  8.89   

       Total Cases per Million New Deaths per Million  \
0                         3.60                    0.0   
1                         3.78                    0.0   
2                         0.01                    0.0   
3                         0.16                    0.0   
4                         3.96                    0.0   
...                        ...                    ...   
31817                 19974.38                    0.0   
31818                 18959.11                    0.0   
31819                 10002.74                    0.0   
31820                 10680.15                    0.0   
31821                 24153.07                    0.1   

      Total Deaths per Million Total Deaths per 100rb Case Fatality Rate  \
0                         1.84                   0.18             51.28%   
1                         1.84                   0.18             48.78%   
2                         0.00                   0.00              0.00%   
3                         0.00                   0.00              0.00%   
4                         1.84                   0.18             46.51%   
...                        ...                    ...                ...   
31817                   459.14                  45.91              2.30%   
31818                   429.59                  42.96              2.27%   
31819                   410.83                  41.08              4.11%   
31820                   221.04  

In [4]:
# New case per hari di Jakarta
case_Jakarta = df[df["Location"] == "DKI Jakarta"][["Date", "New Cases"]].groupby("Date").sum()
# New case per hari di Jawa Barat
case_JB = df[df["Location"] == "Jawa Barat"][["Date", "New Cases"]].groupby("Date").sum()

print(f'Rata - rata new case Jakarta : {round(case_Jakarta["New Cases"].mean(),2)}')
print(f'Rata - rata new case Jawa Barat : {round(case_JB["New Cases"].mean(),2)}')

Rata - rata new case Jakarta : 1520.42
Rata - rata new case Jawa Barat : 1266.16


Rata - rata new case Jakarta : 1520.42

Rata - rata new case Jawa Barat : 1266.16

H0: μ_Jakarta = μ_JawaBarat

H1: μ_Jakarta != μ_JawaBarat

Apakah rata-rata new case Jakarta tidak ada perbedaan signifikan pada rata-rata new case Jawa Barat atau terdapat perbedaan yang signifikan

### Pembuktian dengan single sample two sided

In [5]:
t_stat, p_val = stats.ttest_ind(case_Jakarta,case_JB)
print('P-value:',p_val[0])
print('t-statistics:',t_stat[0])

P-value: 0.020253989345214322
t-statistics: 2.3236112018015596


Hasil P-value adalah 0.02 yaitu dibawah critical value (0.05), maka hipotesis null yang menyatakan tidak ada perbedaan signifikan dapat ditolak. Sehingga rata-rata new case Jakarta dan Jawa Barat berbeda cukup signifikan.